index.d.ts

```diff
@@ -0,0 +1,121 @@
+import fastify, {
+  FastifyError,
+  FastifyReply,
+  FastifyRequest,
+  RegisterOptions
+} from "fastify"
+import { IResolvers } from "graphql-tools";
+import { Server, IncomingMessage, ServerResponse } from "http";
+import graphql, { GraphQLSchema, GraphQLError, Source, DocumentNode, ExecutionResult } from 'graphql';
+
+declare namespace FastifyGQL {
+
+  export interface Plugin {
+    /**
+     * Extend existing schema
+     * @param schema graphql schema
+     */
+    extendSchema(schema: string | Source | DocumentNode): void;
+    /**
+     * Define additional resolvers
+     * @param resolvers object with resolver functions
+     */
+    defineResolvers(resolvers: IResolvers): void;
+    /**
+     * Define data loaders
+     * @param loaders object with data loader functions
+     */
+    defineLoaders(loaders: {[key: string]: Function}): void;
+  }
+
+  export interface Options<
+    HttpServer,
+    HttpRequest,
+    HttpResponse
```

Comment:
In order to support http2, we might have to do: https://github.com/fastify/fastify-swagger/blob/2f7d7f68f8d87f16904b5da8bec38784b64ac5c8/index.d.ts#L70-L72.